dev/bots/analyze_snippet_code.dart

```diff
@@ -188,22 +249,56 @@ class _SnippetChecker {
   static const String _dartDocPrefixWithSpace = '$_dartDocPrefix ';
 
   /// A RegExp that matches the beginning of a dartdoc snippet.
-  static final RegExp _dartDocSnippetBeginRegex = RegExp(r'{@tool snippet(?:| ([^}]*))}');
+  static final RegExp _dartDocSnippetBeginRegex = RegExp(r'{@tool ([^ }]+)(?:| ([^}]*))}');
 
   /// A RegExp that matches the end of a dartdoc snippet.
   static final RegExp _dartDocSnippetEndRegex = RegExp(r'{@end-tool}');
 
   /// A RegExp that matches the start of a code block within dartdoc.
-  static final RegExp _codeBlockStartRegex = RegExp(r'///\s+```dart.*$');
+  static final RegExp _codeBlockStartRegex = RegExp(r'^ */// *```dart$');
+
+  /// A RegExp that matches the start of a code block within a regular comment.
+  /// Such blocks are not not analyzed. They can be used to give sample code for
+  /// internal (private) APIs where visibilty would make analyzing the sample
+  /// code problematic.
+  static final RegExp _uncheckedCodeBlockStartRegex = RegExp(r'^ *// *```dart$');
 
   /// A RegExp that matches the end of a code block within dartdoc.
-  static final RegExp _codeBlockEndRegex = RegExp(r'///\s+```\s*$');
+  static final RegExp _codeBlockEndRegex = RegExp(r'^ */// *``` *$');
+
+  /// A RegExp that matches a line starting with a comment or annotation
+  static final RegExp _nonCodeRegExp = RegExp(r'^ *(//|@)');
+
+  /// A RegExp that matches things that look like a function declaration.
+  static final RegExp _maybeFunctionDeclarationRegExp = RegExp(r'^([A-Z][A-Za-z0-9_<>, ?]*|int|double|bool|void)\?? (_?[a-z][A-Za-z0-9_<>]*)\(.*');
+
+  /// A RegExp that matches things that look like a getter.
+  static final RegExp _maybeGetterDeclarationRegExp = RegExp(r'^([A-Z][A-Za-z0-9_<>?]*|int|double|bool)\?? get (_?[a-z][A-Za-z0-9_<>]*) (?:=>|{).*');
+
+  /// A RegExp that matches an identifier followed by a colon, potentially with two spaces of indent.
+  static final RegExp _namedArgumentRegExp = RegExp(r'^(?:  )?([a-zA-Z0-9_]+): ');
+
+  /// A RegExp that matches things that look unambiguously like top-level declarations.
+  static final RegExp _topLevelDeclarationRegExp = RegExp(r'^(abstract|class|mixin|enum|typedef|final) ');
```

Comment:
Added `extension`, though hopefully no examples even use that.

`const` is intentionally not here because it's used when showing examples of Widgets, which are expressions.

Top-level functions are handled by `_maybeFunctionDeclarationRegExp`.

Mutable variables are hard to recognize. We never use `var` because of the analysis options. Generally they end up being treated as statements, which seems to work for our current examples.